docs/developer/architecture/logs.rst

```diff
@@ -0,0 +1,120 @@
+Log Centralisation
+==================
+
+Context
+-------
+
+MetalK8s value is to provide, out of the box, some services in order to ease
+the monitoring and operation of the platform as well as workload running on top
+of it.
+
+MetalK8s provides monitoring service with Prometheus and AlertManager in
+order to expose MetalK8s and workloads metrics and alerts.
+
+The logs generated by the platform and the workloads constitute an essential
+piece of information that can be very helpful when it comes to understand the
+root cause of a failure or a performance degradation. Because of the
+distributed nature of MetalK8s and workloads running of top of it, it is
+important to have some tooling to ease the analysis of near real time and past
+logs from a central endpoint. As such those logs should be stored on the system
+for a configurable period of time. The logs analysis functionalities should be
+accessible through an API and a UI. It is also important to ease correlation
+with other type of indicators such as KPIs and alerts.
+
+The following requirements are not about audit logs but only focus on
+application logs.
+
+Requirements
+------------
+
+- Accessible from a central UI/API
+
+Platform Admin or Storage Admin can visualize logs from all containers from all
+namespaces as well as kubelet, containerd, salt-minion, metalk8s or CLI logs in
+Grafana. One can correlate logs and metrics or alerts in one single Grafana
+Dashboard.
+Browsing logs can be achieved through a documented API.
+
+- Persistence, Retention
+
+Logs should be stored on a persistent storage. Some automatic purging mechanism
+should ensure we are never filling the persistent storage. The Platform
+Administrator can configure the retention period.
```

Comment:
Hmmm, if I set up a 50Gb PV with a retention period configured to last 10 days and for some reason, my PV is filled in 5 days because one of my applications was started in debug mode.

What then sets in here?
Do we automatically start purging things so as to not fill the volume or we respect the Admins retention period (meaning we might lose logs when Volume is filled)?

Seems like here, we have to prioritize one or this is a design issue and should not be raised here?

Reply:
clarified